### DISCUSSION TITLE
Having issue installing on windows.

### DISCUSSION THREAD

**Discussion by uorufu** [gitmylo/audio-webui#188] [Q&A]
```
Traceback (most recent call last):
  File "D:\audiowebui\audio-webui\main.py", line 44, in <module>
    autodebug.catcher(e)
  File "D:\audiowebui\audio-webui\autodebug\autodebug.py", line 79, in catcher
    raise e
  File "D:\audiowebui\audio-webui\main.py", line 26, in <module>
    from webui.modules.implementations.tts_monkeypatching import patch as patch1
  File "D:\audiowebui\audio-webui\webui\modules\implementations\__init__.py", line 1, in <module>
    import webui.modules.implementations.ttsmodels as tts
  File "D:\audiowebui\audio-webui\webui\modules\implementations\ttsmodels.py", line 14, in <module>
    import webui.modules.models as mod
  File "D:\audiowebui\audio-webui\webui\modules\models.py", line 6, in <module>
    from transformers import Pipeline
  File "D:\audiowebui\audio-webui\venv\lib\site-packages\transformers\__init__.py", line 26, in <module>
    from . import dependency_versions_check
  File "D:\audiowebui\audio-webui\venv\lib\site-packages\transformers\dependency_versions_check.py", line 16, in <module>
    from .utils.versions import require_version, require_version_core
  File "D:\audiowebui\audio-webui\venv\lib\site-packages\transformers\utils\__init__.py", line 61, in <module>
    from .hub import (
  File "D:\audiowebui\audio-webui\venv\lib\site-packages\transformers\utils\hub.py", line 94, in <module>
    PYTORCH_PRETRAINED_BERT_CACHE = os.getenv("PYTORCH_PRETRAINED_BERT_CACHE", constants.HF_HUB_CACHE)
AttributeError: module 'huggingface_hub.constants' has no attribute 'HF_HUB_CACHE'

```
Not sure what I did wrong.

**Comment by gitmylo** [comment]
That's weird, this error seems to be entirely within huggingface_hub. Or it might be incorrectly initialized, this is not an error on your end. This is either caused by a change in huggingface_hub or a difference in your python environment.

**Reply by uorufu** [reply]
I'll try recreate the venv differently or look for answers from their side. I did find something on hugging face about environment variable issues when not installed as admin. I think I did install as admin, but now I am not sure.

**Comment by uorufu** [comment]
> That's weird, this error seems to be entirely within huggingface_hub. Or it might be incorrectly initialized, this is not an error on your end. This is either caused by a change in huggingface_hub or a difference in your python environment.

If I manually declare HF_HUB_CACHE and HF_HOME in constants.py it throws this error

> Traceback (most recent call last):
>   File "D:\ai\audio-webui\main.py", line 26, in <module>
>     from webui.modules.implementations.tts_monkeypatching import patch as patch1
>   File "D:\ai\audio-webui\webui\modules\implementations\__init__.py", line 1, in <module>
>     import webui.modules.implementations.ttsmodels as tts
>   File "D:\ai\audio-webui\webui\modules\implementations\ttsmodels.py", line 14, in <module>
>     import webui.modules.models as mod
>   File "D:\ai\audio-webui\webui\modules\models.py", line 6, in <module>
>     from transformers import Pipeline
>   File "D:\ai\audio-webui\venv\lib\site-packages\transformers\__init__.py", line 26, in <module>
>     from . import dependency_versions_check
>   File "D:\ai\audio-webui\venv\lib\site-packages\transformers\dependency_versions_check.py", line 57, in <module>
>     require_version_core(deps[pkg])
>   File "D:\ai\audio-webui\venv\lib\site-packages\transformers\utils\versions.py", line 117, in require_version_core
>     return require_version(requirement, hint)
>   File "D:\ai\audio-webui\venv\lib\site-packages\transformers\utils\versions.py", line 111, in require_version
>     _compare_versions(op, got_ver, want_ver, requirement, pkg, hint)
>   File "D:\ai\audio-webui\venv\lib\site-packages\transformers\utils\versions.py", line 44, in _compare_versions
>     raise ImportError(
> ImportError: huggingface-hub>=0.19.3,<1.0 is required for a normal functioning of this module, but found huggingface-hub==0.17.1.


That's about as far as I have gotten with it. 

Manually upgrading hf hub in the venv and running with --skip-install doesnt work either and throws the same error, without the arg it just reverts to 0.17.1 and HF_HUB_CACHE and HF_HOME are no longer declared in constants.py and throwing up errors again.
